embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/session/BackgroundActivityDisabledTest.kt

```diff
@@ -0,0 +1,231 @@
+import io.embrace.android.embracesdk.getSentLogs
+import io.embrace.android.embracesdk.getSentSessions
+import io.embrace.android.embracesdk.getSessionId
+import io.embrace.android.embracesdk.internal.arch.schema.EmbType
+import io.embrace.android.embracesdk.internal.clock.nanosToMillis
+import io.embrace.android.embracesdk.internal.opentelemetry.embCleanExit
+import io.embrace.android.embracesdk.internal.opentelemetry.embColdStart
+import io.embrace.android.embracesdk.internal.opentelemetry.embProcessIdentifier
+import io.embrace.android.embracesdk.internal.opentelemetry.embSequenceId
+import io.embrace.android.embracesdk.internal.opentelemetry.embSessionEndType
+import io.embrace.android.embracesdk.internal.opentelemetry.embSessionNumber
+import io.embrace.android.embracesdk.internal.opentelemetry.embSessionStartType
+import io.embrace.android.embracesdk.internal.opentelemetry.embState
+import io.embrace.android.embracesdk.internal.opentelemetry.embTerminated
+import io.embrace.android.embracesdk.internal.payload.Span
+import io.embrace.android.embracesdk.internal.spans.findAttributeValue
+import io.embrace.android.embracesdk.internal.worker.WorkerName
+import io.embrace.android.embracesdk.recordSession
+import io.embrace.android.embracesdk.spans.EmbraceSpan
+import io.opentelemetry.semconv.incubating.SessionIncubatingAttributes
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Verify functionality of the SDK if background activities are disabled
+ */
+@RunWith(AndroidJUnit4::class)
+internal class BackgroundActivityDisabledTest {
+
+    @Rule
+    @JvmField
+    val testRule: IntegrationTestRule = IntegrationTestRule {
+        val clock = FakeClock()
+        val initModule = FakeInitModule(clock)
+        val workerThreadModule = FakeWorkerThreadModule(initModule, WorkerName.REMOTE_LOGGING)
+
+        IntegrationTestRule.Harness(
+            overriddenClock = clock,
+            overriddenInitModule = initModule,
+            overriddenWorkerThreadModule = workerThreadModule,
+        ).apply {
+            overriddenConfigService.backgroundActivityCaptureEnabled = false
+        }
+    }
+
+    @Test
+    fun `recording telemetry in the background when background activity is disabled does the right thing`() {
+        with(testRule) {
+            lateinit var trace: EmbraceSpan
+            harness.recordSession {
+                trace = checkNotNull(embrace.startSpan("test-trace"))
+            }
+            runLoggingThread()
+
+            val traceStopMs = harness.overriddenClock.tick(100L)
+            assertTrue(trace.stop())
+
+            // Check what should and shouldn't be logged when there is no background activity and the app is in the background
```

Comment:
Perhaps worth verifying no session payload got sent to the `DeliveryService`? Either here or implicitly later on by checking the request count

Reply:
Sounds good